tests/posit/connect/test_context.py

```diff
@@ -0,0 +1,90 @@
+from email.contentmanager import ContentManager
+from unittest.mock import MagicMock, Mock
+
+import pytest
+import requests
+import responses
+
+from posit.connect.context import Context, requires
+from posit.connect.urls import Url
+
+
+class TestRequires:
+    def test_version_unsupported(self):
+        class Stub(ContentManager):
+            def __init__(self, ctx):
+                self.ctx = ctx
+
+            @requires("1.0.0")
+            def fail(self):
+                pass
+
+        ctx = MagicMock()
+        ctx.version = "0.0.0"
+        instance = Stub(ctx)
+
+        with pytest.raises(RuntimeError):
+            instance.fail()
+
+    def test_version_supported(self):
+        class Stub(ContentManager):
+            def __init__(self, ctx):
+                self.ctx = ctx
+
+            @requires("1.0.0")
+            def success(self):
+                pass
+
+        ctx = MagicMock()
+        ctx.version = "1.0.0"
+        instance = Stub(ctx)
+
+        instance.success()
+
+    def test_version_missing(self):
+        class Stub(ContentManager):
+            def __init__(self, ctx):
+                self.ctx = ctx
+
+            @requires("1.0.0")
+            def success(self):
+                pass
+
+        ctx = MagicMock()
+        ctx.version = None
+        instance = Stub(ctx)
+
+        instance.success()
```

Comment:
This test is for behavior we use while testing, but not one we expect users to hit, yeah? If yeah, it would be good to comment about that just so it's clear this isn't a desired end-user behavior.

Reply:
No, this is designed behavior. If the version is hidden in the server settings, then `ctx.version` will be None. When the value is `None` the version check is skipped since we aren't sure. In the case of breaking API changes, we will need to perform the fallback sequence based on the HTTP response.